### PR TITLE
Normative: Allow locale based ignorePunctuation default

### PR DESCRIPTION
**Pull request by FrankYFTang**
"th" locale in CLDR collation rule default ignorePunctuation to true instead of false so we need to get tht default value from the locale data instead of always false.
Address https://github.com/tc39/ecma402/issues/832

th CLDR rule is in 
https://github.com/unicode-org/cldr/blob/main/common/collation/th.xml#L17
```
				[alternate shifted]

```

@anba @sffc @ben-allen  @gibson042

### CODE REVIEW

**Review by anba — APPROVED** [review]
This looks reasonable to me from the spec point of view. I haven't yet investigated what needs to be changed implementation-wise, though.

One small nit: Annex A <https://tc39.es/ecma402/#annex-implementation-dependent-behaviour> should also be updated to include:
> The default ignore punctuation value per locale (&lt;xref> here)

File: spec/collator.html [hunk 1]
```diff
@@ -136,6 +137,7 @@ <h1>Internal slots</h1>
         <li>The first element of [[SortLocaleData]].[[&lt;_locale_&gt;]].[[co]] and [[SearchLocaleData]].[[&lt;_locale_&gt;]].[[co]] must be *null*.</li>
         <li>The values *"standard"* and *"search"* must not be used as elements in any [[SortLocaleData]].[[&lt;_locale_&gt;]].[[co]] and [[SearchLocaleData]].[[&lt;_locale_&gt;]].[[co]] list.</li>
         <li>[[SearchLocaleData]].[[&lt;_locale_&gt;]] must have a [[sensitivity]] field with a String value equal to *"base"*, *"accent"*, *"case"*, or *"variant"*.</li>
+        <li>[[SearchLocaleData]].[[&lt;_locale_&gt;]] and [[SortLocaleData]].[[&lt;_locale_&gt;]] must have a [[ignorePunctuation]] field with a Boolean value.</li>
```
Comment (ben-allen) [inline spec/collator.html:140]:
```suggestion
        <li>[[SearchLocaleData]].[[&lt;_locale_&gt;]] and [[SortLocaleData]].[[&lt;_locale_&gt;]] must have an [[ignorePunctuation]] field with a Boolean value.</li>
```

**Review by ben-allen — APPROVED** [review]
Minor typo fix, otherwise great

**Comment by FrankYFTang** [comment]
Changed. PTAL

**Comment by ben-allen** [comment]
LGTM!

**Review by sffc — APPROVED** [review]
This seems like a good change to me; let's discuss in the next TG2

**Review by anba — APPROVED** [review]

**Comment by sffc** [comment]
TG2 approval: https://github.com/tc39/ecma402/blob/master/meetings/notes-2023-12-14.md#normative-allow-locale-based-ignorepunctuation-default-833

**Comment by FrankYFTang** [comment]
This is on the agenda for discussion on 2024-02-06 TC39 meeting https://github.com/tc39/agendas/blob/main/2024/02.md

**Comment by FrankYFTang** [comment]
Has concensus from TG1 this morning around 11:25am PST (2024-02-06)

**Comment by FrankYFTang** [comment]
Please merge

**Pull request merged**